substrate/frame/revive/src/lib.rs

```diff
@@ -2095,6 +2095,21 @@ impl<T: Config> Pallet<T> {
 			.map_err(ContractAccessError::StorageWriteFailed)
 	}
 
+	/// Uploads evm runtime code and returns the Vm binary contract blob and deposit amount
+	/// collected.
+	pub fn try_upload_evm_runtime_code(
+		origin: H160,
+		code: Vec<u8>,
+		storage_deposit_limit: BalanceOf<T>,
+		exec_config: &ExecConfig,
+	) -> Result<(ContractBlob<T>, BalanceOf<T>), DispatchError> {
+		let origin = T::AddressMapper::to_account_id(&origin);
+		let mut module = ContractBlob::from_evm_runtime_code(code, origin)?;
+		let deposit = module.store_code(exec_config, None)?;
+		ensure!(storage_deposit_limit >= deposit, <Error<T>>::StorageDepositLimitExhausted);
+		Ok((module, deposit))
+	}
```

Comment:
can't the exisiting try_upload_pvm_code be renamed try_upload_code and accept both?

Reply:
here origin is H160 instead of AccountId, but we can use AccountId in both cases and have one method if you want